From: Mike Hommey <mh+mozilla@glandium.org>
Date: Sat, 15 Aug 2015 08:53:54 +0900
Subject: Bug 1194905 - Build libvpx neon code without -mthumb and
 -mfloat-abi=softfp

-mfloat-abi=softfp is hardcoding an ABI that may not match everything else
in Gecko, which is bound to fail. It's also unrelated to building NEON code.
-mthumb is also unrelated to building NEON code, and should be left as it
is for everything else.
Now, interestingly, we also have a VPX_ASFLAGS variable that just contains
the right things instead of hardcoding them in the moz.build, so just use
that instead.
---
 media/libvpx/moz.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/libvpx/moz.build b/media/libvpx/moz.build
index a1138f4..a8298d4 100644
--- a/media/libvpx/moz.build
+++ b/media/libvpx/moz.build
@@ -65,7 +65,7 @@ if CONFIG['VPX_ARM_ASM']:
 
     for f in SOURCES:
         if f.endswith('.c') and 'neon' in f:
-            SOURCES[f].flags += ['-march=armv7-a', '-mthumb', '-mfloat-abi=softfp', '-mfpu=neon']
+            SOURCES[f].flags += CONFIG['VPX_ASFLAGS'].split()
 
 # boolhuff_armv5te.asm defines the same functions as boolhuff.c instead of
 # using RTCD, so we have to make sure we only add one of the two.
